api/core/app/apps/message_based_app_generator.py

```diff
@@ -156,78 +156,85 @@ def _init_generate_records(
                 from_end_user_id=end_user_id,
                 from_account_id=account_id,
+                app_mode=app_config.app_mode,
             )
 
-            db.session.add(conversation)
-            db.session.commit()
-            db.session.refresh(conversation)
-        else:
-            conversation.updated_at = naive_utc_now()
-            db.session.commit()
+            db.session.add(message)
+            db.session.flush()
+            db.session.refresh(message)
 
-        message = Message(
-            app_id=app_config.app_id,
-            model_provider=model_provider,
-            model_id=model_id,
-            override_model_configs=json.dumps(override_model_configs) if override_model_configs else None,
-            conversation_id=conversation.id,
-            inputs=application_generate_entity.inputs,
-            query=application_generate_entity.query,
-            message="",
-            message_tokens=0,
-            message_unit_price=0,
-            message_price_unit=0,
-            answer="",
-            answer_tokens=0,
-            answer_unit_price=0,
-            answer_price_unit=0,
-            parent_message_id=getattr(application_generate_entity, "parent_message_id", None),
-            provider_response_latency=0,
-            total_price=0,
-            currency="USD",
-            invoke_from=application_generate_entity.invoke_from.value,
-            from_source=from_source,
-            from_end_user_id=end_user_id,
-            from_account_id=account_id,
-            app_mode=app_config.app_mode,
-        )
+            message_files = []
+            for file in application_generate_entity.files:
+                message_file = MessageFile(
+                    message_id=message.id,
+                    type=file.type,
+                    transfer_method=file.transfer_method,
+                    belongs_to="user",
+                    url=file.remote_url,
+                    upload_file_id=file.related_id,
+                    created_by_role=(CreatorUserRole.ACCOUNT if account_id else CreatorUserRole.END_USER),
+                    created_by=account_id or end_user_id or "",
+                )
+                message_files.append(message_file)
 
-        db.session.add(message)
-        db.session.commit()
-        db.session.refresh(message)
-
-        for file in application_generate_entity.files:
-            message_file = MessageFile(
-                message_id=message.id,
-                type=file.type,
-                transfer_method=file.transfer_method,
-                belongs_to="user",
-                url=file.remote_url,
-                upload_file_id=file.related_id,
-                created_by_role=(CreatorUserRole.ACCOUNT if account_id else CreatorUserRole.END_USER),
-                created_by=account_id or end_user_id or "",
-            )
-            db.session.add(message_file)
-            db.session.commit()
+            if message_files:
```

Comment:
IMO, only removing the `db.session.commit` inside for loop is enough, since `db.session.add` only adds the newly created `MessageFile` object to session and does not have any IO operation.